### PR TITLE
Implement str.replace

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

**Comment by Rubtsowa** [comment]
Please, review this PR.

**Comment by Rubtsowa** [comment]
Сhange the label, please.

**Comment by esc** [comment]
@Rubtsowa thanks for the ping on this PR, I have added it to the queue for review.

File: numba/tests/test_unicode.py [hunk 1]
```diff
@@ -950,6 +958,9 @@ def test_join(self):
         ]
 
         for sep, parts in CASES:
+            print(sep)
```
Comment (stuartarchibald) [inline numba/tests/test_unicode.py]: please remove this print statements.

**Comment by Rubtsowa** [comment]
Please, review this PR. 
conflicts resolved

**Comment by Rubtsowa** [comment]
Please check this PR! If everything is correct, please merge.

**Pull request closed without merging**